include/swift/AST/DiagnosticsSema.def

```diff
@@ -1335,7 +1335,7 @@ ERROR(member_shadows_global_function,none,
       (DeclNameRef, DescriptiveDeclKind, DescriptiveDeclKind, DeclName, DeclName))
 
 ERROR(instance_member_use_on_type,none,
-      "instance member %1 cannot be used on type %0; "
+      "instance member %1 of type %0 cannot be used in static context; "
```

Comment:
I don't think that "in static context" brings any clarify because everybody has a different idea what "static context" means, but I'm biased. @hborla WDYT?

Reply:
I’m probably biased too, but I read this as "you cannot use this at all in a static function". Regrettably, the best alternative I can currently think of is `instance member cannot be used on type object`.

Reply:
However, maybe it does make sense to say "static context" specifically when `self.` is implicit.

Reply:
> "you cannot use this at all in a static function".

This is exactly what I'm afraid of happening.

Reply:
Guys, how should we proceed with this fix?

Reply:
The primary goal is to tailor the message for implicit accesses on `self` specifically. The current message reads well when applied to an explicit access on a type identifier as in `T.member()`. As for how exactly we want to tailor it, we need to come up with a better idea than what was voiced in the issue.

Reply:
Maybe a good start would be to limit this diagnostic to situations when `self.` is implicit (that's what "in static context" means for a lot of people IMHO).

Reply:
Putting in my two cents: `instance member %1 cannot be used on static 'self'`